src/tls/test/main.cpp

```diff
@@ -0,0 +1,378 @@
+  int read = server.read(buf, message_length);
+  REQUIRE(read == message_length);
+  buf[message_length] = '\0';
+  test_log.write(
+    fmt::format("Server message received [{}]\n", string((const char*)buf)));
+  REQUIRE(strncmp((const char*)buf, (const char*)message, message_length) == 0);
+
+  // Send the response
+  test_log.write(fmt::format(
+    "Server sending message [{}]\n", string((const char*)response)));
+  written = server.write(response, response_length);
+  REQUIRE(written == response_length);
+
+  // Receive the response
+  read = client.read(buf, response_length);
+  REQUIRE(read == response_length);
+  buf[response_length] = '\0';
+  test_log.write(
+    fmt::format("Client message received [{}]\n", string((const char*)buf)));
+  REQUIRE(
+    strncmp((const char*)buf, (const char*)response, response_length) == 0);
+
+  test_log.write("Closing connection\n");
+  client.close();
+  server.close();
+}
+
+TEST_CASE("unverified handshake")
+{
+  // Create a CA
+  auto ca = get_ca();
+
+  // Create bogus certificate
+  auto server_cert = get_dummy_cert(ca, "server");
+  auto client_cert = get_dummy_cert(ca, "client");
+
+  // Just testing handshake, does not verify certificates, no communication.
+  run_test_case(
+    0,
+    (const uint8_t*)"",
+    0,
+    (const uint8_t*)"",
+    0,
+    move(server_cert),
+    move(client_cert),
+    false);
+}
+
+TEST_CASE("unverified communication")
+{
+  const uint8_t message[] = "Hello World!";
+  size_t message_length = strlen((const char*)message);
+  const uint8_t response[] = "Hi back!";
+  size_t response_length = strlen((const char*)response);
+
+  // Create a CA
+  auto ca = get_ca();
+
+  // Create bogus certificate
+  auto server_cert = get_dummy_cert(ca, "server");
+  auto client_cert = get_dummy_cert(ca, "client");
+
+  // Just testing communication channel, does not verify certificates.
+  run_test_case(
+    0,
+    message,
+    message_length,
+    response,
+    response_length,
+    move(server_cert),
+    move(client_cert),
+    false);
+}
+
+TEST_CASE("verified handshake")
+{
+  // Create a CA
+  auto ca = get_ca();
+
+  // Create bogus certificate
+  auto server_cert = get_dummy_cert(ca, "server");
+  auto client_cert = get_dummy_cert(ca, "client");
+
+  // Just testing handshake, no communication, but verifies certificates.
+  run_test_case(
+    0,
+    (const uint8_t*)"",
+    0,
+    (const uint8_t*)"",
+    0,
+    move(server_cert),
+    move(client_cert),
+    true);
+}
+
+TEST_CASE("verified communication")
+{
+  const uint8_t message[] = "Hello World!";
+  size_t message_length = strlen((const char*)message);
+  const uint8_t response[] = "Hi back!";
+  size_t response_length = strlen((const char*)response);
+
+  // Create a CA
+  auto ca = get_ca();
+
+  // Create bogus certificate
+  auto server_cert = get_dummy_cert(ca, "server");
+  auto client_cert = get_dummy_cert(ca, "client");
+
+  // Just testing communication channel, does not verify certificates.
```

Comment:
This is the same comment than on line `321`, but for a different test. Should this be updated?